docs/en/creating-widgets/supplemental.md

```diff
@@ -0,0 +1,891 @@
+
+Applications only need to concern themselves with declaring their intended output structure as a hierarchy of virtual DOM nodes, typically done as the return values from their [widgets' render functions](#basic-widget-structure). The framework's [`Renderer`](#rendering-to-the-dom) component then synchronizes the intended output with concrete elements in the DOM. Virtual DOM nodes also serve to configure and provide state to widgets and elements, by passing in properties.
+
+Dojo supports subtree rendering, meaning that when a change in state occurs, the framework is able to determine specific subsets of VDOM nodes affected by the change. Only the required corresponding subtrees within are then updated to reflect the change, increasing rendering performance and resulting in an improved user interactive experience.
+
+> **Note:** Returning virtual nodes from widget render functions is the only concern applications have around rendering. Attempting to use any other practice [is considered an anti-pattern in Dojo application development](#widget-development-best-practices), and should be avoided.
+
+## TSX Support
+
+Dojo supports use of the `jsx` syntax extension known as [`tsx` in TypeScript](https://www.TypeScriptlang.org/docs/handbook/jsx.html). This syntax allows for a more convenient representation of a widget's VDOM output that is closer to the resulting HTML within a built application.
+
+### TSX-enabled applications
+
+TSX-enabled projects can easily be scaffolded via the [`dojo create app --tsx` CLI command](https://github.com/dojo/cli-create-app).
+
+For Dojo projects that were not scaffolded in this way, TSX can be enabled with the following additions to the project's TypeScript config:
+
+> `./tsconfig.json`
+
+```json
+{
+	"compilerOptions": {
+		"jsx": "react",
+		"jsxFactory": "tsx"
+	},
+	"include": ["./src/**/*.ts", "./src/**/*.tsx"]
+}
+```
+
+### TSX widget example
+
+Widgets with a `.tsx` file extension can output TSX from their render function by simply importing the `tsx` function from the `@dojo/framework/core/vdom` module:
+
+> src/widgets/MyTsxWidget.tsx
+
+**Function-based variant:**
+
+```tsx
+import { create, tsx } from '@dojo/framework/core/vdom';
+
+const factory = create();
+
+export default factory(function MyTsxWidget() {
+	return <div>Hello from a TSX widget!</div>;
+});
+```
+
+**Class-based variant:**
+
+```tsx
+import WidgetBase from '@dojo/framework/core/WidgetBase';
+import { tsx } from '@dojo/framework/core/vdom';
+
+export default class MyTsxWidget extends WidgetBase {
+	protected render() {
+		return <div>Hello from a TSX widget!</div>;
+	}
+}
+```
+
+Widgets that need to return multiple top-level TSX nodes can wrap them in a `<virtual>` container element. This is a clearer option than returning an array of nodes as it allows for more natural automated code formatting within TSX blocks. For example:
```

Comment:
It's also possible to use `v('virtual',` instead of returning an array - wonder whether we should mention that or not.